internal/bsr/bsr.go

```diff
@@ -402,9 +431,21 @@ func (c *Connection) OpenChannel(ctx context.Context, chanId string) (*Channel,
 		return nil, err
 	}
 
+	af, ok := SummaryAllocFuncs.Get(c.session.Meta.Protocol, ChannelContainer)
+	if !ok {
+		return nil, fmt.Errorf("%s: failed to get summary type", op)
```

Comment:
Do we have any unhappy path unit testing to validate this check?

Reply:
I just added unit tests to cover this:
https://github.com/hashicorp/boundary/blob/d668531c764f7d2f59d30c5a13786c338a8ecbc1/internal/bsr/bsr_open_test.go#L331